consumer.go

```diff
@@ -118,25 +123,39 @@ func NewConsumerWithOptions(options *ConsumerOptions) (*Consumer, error) {
 	return &Consumer{
 		Errors: make(chan error),
 
-		options: options,
-		redis:   r,
-		funcs:   map[string]ConsumerFunc{},
-		streams: make([]string, 0),
-		queue:   make(chan *Message, options.BufferSize),
-		wg:      &sync.WaitGroup{},
+		options:   options,
+		redis:     r,
+		consumers: make(map[string]registeredConsumer),
+		streams:   make([]string, 0),
+		queue:     make(chan *Message, options.BufferSize),
+		wg:        &sync.WaitGroup{},
 
 		stopReclaim: make(chan struct{}, 1),
 		stopPoll:    make(chan struct{}, 1),
 		stopWorkers: make(chan struct{}, options.Concurrency),
 	}, nil
 }
 
+// RegisterWithLastID is the same as Register, except that it also lets you
+// specify the oldest message to receive. This can be any valid message ID, "0"
+// for all messages in the stream, or "$" for only new messages.
```

Comment:
Let's add a note here saying that the `id` value is only used when the consumer group is created for the first time, and if the consumer group already exists, the value of `id` will be ignored.

Reply:
@robinjoseph08 I've pushed an amended commit which I hope reads okay.